File: android/app/src/main/java/net/globulus/simisync/PrefUtil.java
package net.globulus.simisync;

import android.content.Context;
import android.content.SharedPreferences;

import net.globulus.easyprefs.annotation.Pref;
import net.globulus.easyprefs.annotation.PrefClass;
import net.globulus.easyprefs.annotation.PrefMaster;

@PrefClass(staticClass = false, origin = true)
public class PrefUtil {
    @Pref(nullable = true) String cookie = null;

    @PrefMaster
    public static SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences("beers", Context.MODE_PRIVATE);
    }
}

File: android/app/src/main/java/net/globulus/simisync/PrefUtil.kt
package net.globulus.simisync

import net.globulus.easyprefs.annotation.Pref
import net.globulus.easyprefs.annotation.PrefClass

@PrefClass(staticClass = false, origin = true)
class PrefUtil {
    @Pref(nullable = true) var cookie: String? = null
}
